### PR TITLE
ACC-985: Add support for authentication endpoints in the namespace of a contentgrid app

### PR DESCRIPTION
**Pull request by vierbergenlars**
- **Add extension to add endpoints to runtime contentgrid applications**
- **Add AuthenticationJwtClaims resolver with extra information for authentication endpoints**
- **Support loading symmetric keys for signing and encryption. (Also support loading all key types from JWK files instead of only PEM)**
- **Fall back to main token filter when no custom signer is available for an endpoint**

### CODE REVIEW

File: src/main/java/com/contentgrid/gateway/runtime/RuntimeConfiguration.java [hunk 1]
```diff
@@ -75,32 +81,71 @@ RouteLocator runtimeRouteLocator(
...
+                    .and()
+                    .path("/.contentgrid/"+endpointId+"/**")
+                    .filters(f -> f
+                            .preserveHostHeader()
+                            .removeRequestHeader("Cookie")
+                            .filter(jwtSignerRegistry.hasSigner(endpointId)?new LocallyIssuedJwtGatewayFilter(new SignedJwtIssuer(
+                                    jwtSignerRegistry.getRequiredSigner(endpointId),
+                                    jwtClaimsResolverLocator.findClaimsResolver(endpointId)
+                                            .orElseThrow(() -> new IllegalStateException("No claims resolver found with name '%s'".formatted(endpointId)))
+                            )): tokenFilter)
```
Comment (tgeens) [inline src/main/java/com/contentgrid/gateway/runtime/RuntimeConfiguration.java:108]: formatting could be improved here

**Review by tgeens — APPROVED** [review]

**Comment by tgeens** [comment]
I'm not sure the SonarCloud complaints on `new Random()` usage have any merrit or not

**Comment by thijslemmens** [comment]
So as discussed, we'll keep this branch open for now. There are some pieces of untested crypto code we're not sure we will keep.

**Review by thijslemmens — APPROVED** [review]

**Comment by thijslemmens** [comment]
build is failing

**Review by thijslemmens — APPROVED** [review]

**Pull request merged**